packages/network/src/index.ts

```diff
@@ -62,10 +66,35 @@ export class StacksMainnet implements StacksNetwork {
 
   readonly coreApiUrl: string;
 
-  constructor(networkUrl: NetworkConfig = { url: HIRO_MAINNET_DEFAULT }) {
-    this.coreApiUrl = networkUrl.url;
+  constructor(networkConfig: NetworkConfig) {
+    this.coreApiUrl = networkConfig.url;
   }
 
+  static fromName = (networkName: StacksNetworkName): IStacksNetwork => {
+    switch (networkName) {
+      case 'mainnet':
+        return new StacksMainnet();
+      case 'testnet':
+        return new StacksTestnet();
+      default:
+        throw new Error(
+          `Invalid network name provided. Must be one of the following: ${StacksNetworks.join(
+            ', '
+          )}`
+        );
+    }
+  };
+
+  static fromNameOrNetwork = (network: StacksNetworkName | IStacksNetwork) => {
+    if (StacksNetworks.includes(network as StacksNetworkName)) {
+      // network is StacksNetworkName
+      return StacksNetwork.fromName(network as StacksNetworkName);
+    }
+
+    // network is IStacksNetwork
```

Comment:
this comment doesn't add any more detail than the code it describes

Reply:
Thanks for bringing this up. Sadly, TS `match`ing on types only works in funky ways. I added some of these comments to explicitly show which type we have at this point in code. Firstly, for when it's not immediately clear what the type is to the dev, but TS knows, e.g. using `'senderKey' in txOptions`. Secondly, for when even TS can't accurately detect the type, even if it's clear.

Not sure if this is a good idea or not — comments can also become stale and potentially reflect incorrect types at some point.

In any case, I'll refactor this to better utilise TS.